src/test/java/teammates/test/cases/browsertests/AdminHomePageUiTest.java

```diff
@@ -83,15 +85,15 @@ private void testCreateInstructorAction() throws Exception {
         InstructorAttributes instructorInBackend = getInstructorWithRetry(demoCourseId, instructor.email);
         assertEquals("Instructor Details must have 3 columns", homePage.getMessageFromResultTable(1));
 
-        String encryptedKey = getKeyForInstructorWithRetry(demoCourseId, instructor.email);
-        // use AppUrl from Config because the join link takes its base URL from build.properties
-        String expectedjoinUrl = Config.getAppUrl(Const.ActionURIs.INSTRUCTOR_COURSE_JOIN)
-                                        .withRegistrationKey(encryptedKey)
-                                        .withInstructorInstitution(institute)
-                                        .toAbsoluteString();
         assertEquals("Instructor AHPUiT Instrúctör WithPlusInEmail has been successfully created " + Const.JOIN_LINK,
                 homePage.getMessageFromResultTable(2));
-        assertEquals(expectedjoinUrl, homePage.getJoinLink(homePage.getMessageFromResultTable(2)));
+
+        String encryptedKey = getKeyForInstructorWithRetry(demoCourseId, instructor.email);
+        String actualJoinLink = homePage.getJoinLink(homePage.getMessageFromResultTable(2));
```

Comment:
Can we parse the acutalJoinLink and use `getQuery()` to get the parameters and compare `getQuery()` in `expectedjoinUrl`?

Reply:
Good idea. The code looks much cleaner and has better checks